python/pyarrow/tests/test_parquet.py

```diff
@@ -2009,6 +2021,37 @@ def test_filters_inclusive_set(tempdir, use_legacy_dataset):
     assert False not in result_df['boolean'].values
 
 
+@pytest.mark.pandas
+@parametrize_legacy_dataset
+def test_filters_no_partition_found(tempdir, use_legacy_dataset):
+    fs = LocalFileSystem._get_instance()
+    base_path = tempdir
+
+    integer_keys = [0, 1]
+    string_keys = ['a', 'b', 'c']
+    boolean_keys = [True, False]
+    partition_spec = [
+        ['integer', integer_keys],
+        ['string', string_keys],
+        ['boolean', boolean_keys]
+    ]
+
+    df = pd.DataFrame({
+        'integer': np.array(integer_keys, dtype='i4').repeat(15),
+        'string': np.tile(np.tile(np.array(string_keys, dtype=object), 5), 2),
+        'boolean': np.tile(np.tile(np.array(boolean_keys, dtype='bool'), 5),
+                           3),
+    }, columns=['integer', 'string', 'boolean'])
+
+    _generate_partition_directories(fs, base_path, partition_spec, df)
+
+    with pytest.raises(ValueError, match=r"No partition is found."):
```

Comment:
This only raises an error for the legacy dataset, for the new implementation, we return an empty table. So you will need to add a `if use_legacy_dataset: ...` here (see a similar case in the `test_filters_invalid_pred_op` test just below)

Reply:
I removed this change. Will have a separate JIRA.